test/functions/loss/softmax_cross_entropy_test.rb

```diff
@@ -268,22 +237,16 @@ def check_invalid_reduce(x, t)
     }
   end
 
-  data(data)
-  def test_invalid_reduce(data)
-    _setup(data)
+  def test_invalid_reduce()
     check_invalid_reduce(@x, @t)
   end
 end
 
 class TestNonDefaultIgnoreLabel < Test::Unit::TestCase
-  reduce       = ['mean', 'no']
-  class_weight = [nil, xm::SFloat.ones([3])]
-
-  data = reduce.product(class_weight).collect {|v|
-    {reduce: v[0], class_weight: v[1]}}
-  data = (1..data.size).to_a.zip(data).to_h
+  data(:reduce,       ['mean', 'no'],              group: :reduce, keep: true)
+  data(:class_weight, [nil, xm::SFloat.ones([3])], group: :reduce, keep: true)
```

Comment:
ditto.

Reply:
Thank you for pointing out.
I fixed it.